### PR TITLE
Native authentication extensibility

### PR DESCRIPTION
**Pull request by Yuki-YuXin**
Please check 
https://identitydivision.visualstudio.com/Engineering/_git/devexdub-android-msal-common-native-auth-private-clone/pullrequest/10865?path=/common4j/src/main/com/microsoft/identity/common/java/controllers/results/INativeAuthCommandResult.kt
for detail

### CODE REVIEW

File: common4j/src/main/com/microsoft/identity/common/java/controllers/results/INativeAuthCommandResult.kt [hunk 1]
```diff
@@ -29,10 +29,13 @@ import com.microsoft.identity.common.java.logging.DiagnosticContext
  * INativeAuthCommandResult interface defines the base class for errors used in Native Auth.
  */
 interface INativeAuthCommandResult {
-    data class Redirect(val correlationId: String = DiagnosticContext.INSTANCE.threadCorrelationId) :
-        SignInStartCommandResult, SignInWithSLTCommandResult, SignInSubmitCodeCommandResult,
-        SignInResendCodeCommandResult, SignInSubmitPasswordCommandResult,
-        SignUpStartCommandResult, SignUpSubmitCodeCommandResult,
+    data class Redirect(
+        override val error: String = "browser_required",
```
Comment (SaurabhMSFT) [inline common4j/src/main/com/microsoft/identity/common/java/controllers/results/INativeAuthCommandResult.kt]: What other errors other than "browser_required" is possible for `Redirect` ?

Comment (Yuki-YuXin) [reply common4j/src/main/com/microsoft/identity/common/java/controllers/results/INativeAuthCommandResult.kt]: I personally think Redirect = browser_rquired, but using different naming.

Comment (SaurabhMSFT) [reply common4j/src/main/com/microsoft/identity/common/java/controllers/results/INativeAuthCommandResult.kt]: If "browser_required" is the only possible value passed to the constructor of Redirect class then it should not be part of the constructor. We should take it as a constant value and use that constant as a parameter for calling Error class.

Comment (Yuki-YuXin) [reply common4j/src/main/com/microsoft/identity/common/java/controllers/results/INativeAuthCommandResult.kt]: How about 

```
    data class Redirect(
        override val correlationId: String = DiagnosticContext.INSTANCE.threadCorrelationId,
    ) : Error(error = error, errorDescription = errorDescription, correlationId = correlationId),
        SignInStartCommandResult, SignInWithSLTCommandResult, SignInSubmitCodeCommandResult, SignInResendCodeCommandResult,
        SignInSubmitPasswordCommandResult, SignUpStartCommandResult, SignUpSubmitCodeCommandResult,
        SignUpResendCodeCommandResult, SignUpSubmitPasswordCommandResult,
        SignUpSubmitUserAttributesCommandResult,
        ResetPasswordStartCommandResult, ResetPasswordSubmitCodeCommandResult,
        ResetPasswordResendCodeCommandResult {
            companion object {
                private const val error: String = "browser_required"
                private const val errorDescription: String = "The client's authentication capabilities are insufficient. Please redirect to the browser to complete authentication"
            }
        }
```

Comment (SaurabhMSFT) [reply common4j/src/main/com/microsoft/identity/common/java/controllers/results/INativeAuthCommandResult.kt]: yes, that's better but the constants should have name in all caps and their name should reflect the value contained. Their names could be `BROWSER_REQUIRED_ERROR` and `BROWSER_REQUIRED_ERROR_DESCRIPTION`

**Review by SaurabhMSFT — APPROVED** [review]

**Review by GBakolas — APPROVED** [review]

**Review by SammyO — APPROVED** [review]
Approved, but let's wait with merging until Saurabh's tests are

**Pull request merged**